internal/adapter/sqlite/note_dao.go

```diff
@@ -519,20 +516,26 @@ func (d *NoteDAO) findRows(opts core.NoteFindOpts, selection noteSelection) (*sq
 		return nil
 	}
 
-	if !opts.Match.IsNull() {
+	if 0 < len(opts.Match) {
```

Comment:
Doing that breaks seems to break `case core.MatchStrategyFts:` as we add duplicate `joinClauses`, probably cleaner to keep it all local rather than pop the fts query construction out separately?

```
    assert.go:28: Expected `ambiguous column name: fts_match.notes_fts` (type sqlite3.Error) to be nil
    assert.go:45: Received (type []string):
        []string{}
    assert.go:46:
        ---
    assert.go:47: But expected (type []string):
        []string{"log/2021-01-04.md"}
    assert.go:48:
        ---
    assert.go:49: Diff:
    assert.go:51:       "[]string[0] != []string[1]"
````

Reply:
Ha yes, I missed that.